### ISSUE TITLE
Use derivatives for gradient descent

### ISSUE THREAD

**Issue by gxavier38** [gxavier38/Straw-Truss-Project#3]
The truss generator currently uses stochastic gradient descent.
Use computed gradients to improve learning rate.
